testing/unittests/metrics/MetricFrameworkTests.cpp

```diff
@@ -308,6 +309,62 @@ class MetricFrameworkTests : public CppUnit::TestFixture
         CPPUNIT_ASSERT_EQUAL(0, static_cast<int>(pCounterGauge->queryValue()));
     }
 
+    void Test_historgram_metric()
+    {
+        std::vector<BucketInfo<unsigned>> buckets = {{"le 2", 2}, {"le 4", 4}, {"le 8", 8} };
+        std::shared_ptr<HistogramMetric<unsigned>> pHistogram = std::make_shared<HistogramMetric<unsigned>>("requests.dist", "description", SMeasureCount, buckets);
+
+        HistogramMetricValues<unsigned> values;
+
+        //
+        // Verify that the correct number of buckets is returned
+        values = pHistogram->queryValues();
```

Comment:
To mimic use from a sink this test needs to use an IMetric to retrieve the values.

Reply:
Yes, the sink will need to know it is a histogram.  I don't have a strong opinion on the interface, but the minimum would be something like

__uint64 queryValue();// returning the sum of the values and
std::vector<__uint64> getHistogramValues(); // returning the counts.  (The total count could be returned by a separate function, or caulcated by summing all these values.)